src/components/TodoContainer.js

```diff
@@ -1,98 +1,54 @@
-import axios from 'axios'
 import React, { useState, useEffect } from 'react'
 
 import TodosList from './TodoList'
 import Header from './Header'
 import InputTodo from './InputTodo'
 
+import { getTodosFromApi, deleteFromApi, addTodoToApi } from '../services/api'
+
 export const TodoContainer = () => {
-  const [state, setState] = useState({
-    data: {
-      todos: [],
-    },
-  })
+  const [todos, setTodos] = useState([{ id: 0, title: '', completed: false }])
+
+  /**
+   * Get 10 todo items from API when component is mounted
+   */
+  useEffect(() => {
+    const fetchTodos = async () => {
+      const todos = await getTodosFromApi();
+      setTodos(todos)
+    };
+    fetchTodos();
+  }, [])
 
   // TODO if possible test with REACT testing library integration test
   const onChangeHandle = (id) => {
-    setState({
-      data: {
-        todos: state.data.todos.map((todo) => {
-          if (todo.id === id) {
-            todo.completed = !todo.completed
-          }
-          return todo
-        }),
-      },
-    })
+    setTodos(todos.map((todo) => {
+      if (todo.id === id) {
+        return {...todo, completed: !todo.completed}
+      } else {
+        return todo;
+      }
+    }));
   }
 
   const deleteItem = async (id) => {
-    await axios.delete(`https://jsonplaceholder.typicode.com/todos/${id}`)
-
-    setState({
-      data: {
-        todos: [
-          ...state.data.todos.filter((todo) => {
-            return todo.id !== id
-          }),
-        ],
-      },
-    })
+    await deleteFromApi(id);
+    setTodos(todos.filter((todo) => todo.id !== id))
   }
 
   const addTodoItem = async (title) => {
-    const { data } = await axios.post(
-      'https://jsonplaceholder.typicode.com/todos',
-      {
-        title: title,
-        completed: false,
-      },
-    )
-
-    setState({
-      data: {
-        todos: [
-          ...state.data.todos,
-          {
-            ...data,
-            userId: 1,
-            id: 11,
-          },
-        ],
-      },
-    })
+    const updatedTodo = { id: 11, title, completed: false };
```

Comment:
I'd pass the whole `todo` object here to update if it's a put, otherwise make it a patch? dunno really